web-components/icon/Icon.svelte

```diff
@@ -0,0 +1,74 @@
+<svelte:options tag="leo-icon" />
+
+<script context="module" lang="ts">
+  import './library.default'
+  const parser = new DOMParser()
+
+  const iconRequests = new Map<
+    string,
+    Promise<{ ok: boolean; svg: string | null }>
+  >()
+  const fetchIcon = async (library: IconLibrary, name: string) => {
+    const url = library.resolve(name)
+    const response = await fetch(url)
+    if (!response.ok) return { ok: false, svg: null }
+
+    const parsed = parser.parseFromString(await response.text(), 'text/html')
+    const svgRoot = parsed.querySelector('svg')
+    if (!svgRoot) return { ok: true, svg: '' }
+
+    // If the library has a mutate function, run it.
+    library.mutate?.(svgRoot, name)
```

Comment:
I haven't seen this syntax before! How does this work?

Reply:
It's null propagation - essentially only call the method if it's not null

`ts
const f = foo?.bar?.baz

// is the same as
const f = foo
	? foo.bar
		? foo.bar.baz
        : undefined
    : undefined

// you can also use it for arrays/funcs but the syntax is a bit more funky

const nthItem = items?.[n]
// which is the same as 
const nthItem = items ? items[n] : undefined

// and for functions, they'll only be called if they're defined
const result = func?.(stuff, things)

// which is
const result = func ? func(stuff, things) : undefined

Reply:
Super cool.